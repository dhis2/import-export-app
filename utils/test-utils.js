import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
