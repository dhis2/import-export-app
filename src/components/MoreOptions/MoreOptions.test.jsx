import React from 'react'
import { render } from 'test-utils'

import { MoreOptions } from '../index.js'

const renderMoreOptions = (initiallyVisible, props) =>
    render(
        <MoreOptions {...props} initiallyVisible={initiallyVisible}>
            <p>Child paragraph</p>
        </MoreOptions>
    )

const props = {
    dataTest: 'more-options',
}

describe('matches snapshot', () => {
    it('when not toggled', () => {
        const { asFragment } = renderMoreOptions(false, props)
        expect(asFragment()).toMatchSnapshot()
    })

    it('when toggled', () => {
        const { asFragment } = renderMoreOptions(true, props)
        expect(asFragment()).toMatchSnapshot()
    })

    it('when toggled and different label', () => {
        const { asFragment } = renderMoreOptions(true, {
            ...props,
            label: 'Details',
        })
        expect(asFragment()).toMatchSnapshot()
    })
})
