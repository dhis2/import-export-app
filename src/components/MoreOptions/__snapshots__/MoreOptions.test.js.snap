// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when not toggled 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-test="more-options"
  >
    <header
      class="header"
      data-test="more-options-header"
    >
      <div
        class="chevronHidden"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.293 6.293a1 1 0 011.32-.083l.094.083 5 5a1 1 0 01.083 1.32l-.083.094-5 5a1 1 0 01-1.497-1.32l.083-.094L13.585 12 9.293 7.707a1 1 0 01-.083-1.32z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h2
        class="label"
      >
        Advanced options
      </h2>
    </header>
    <div
      class="children"
      data-test="more-options-children"
    />
  </section>
</DocumentFragment>
`;

exports[`matches snapshot when toggled 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-test="more-options"
  >
    <header
      class="header"
      data-test="more-options-header"
    >
      <div
        class="chevronVisible"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.293 6.293a1 1 0 011.32-.083l.094.083 5 5a1 1 0 01.083 1.32l-.083.094-5 5a1 1 0 01-1.497-1.32l.083-.094L13.585 12 9.293 7.707a1 1 0 01-.083-1.32z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h2
        class="label"
      >
        Advanced options
      </h2>
    </header>
    <div
      class="jsx-1555285825 jsx-876534464 "
      data-test="dhis2-uicore-divider"
    />
    <div
      class="children"
      data-test="more-options-children"
    >
      <p>
        Child paragraph
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches snapshot when toggled and different label 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-test="more-options"
  >
    <header
      class="header"
      data-test="more-options-header"
    >
      <div
        class="chevronVisible"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.293 6.293a1 1 0 011.32-.083l.094.083 5 5a1 1 0 01.083 1.32l-.083.094-5 5a1 1 0 01-1.497-1.32l.083-.094L13.585 12 9.293 7.707a1 1 0 01-.083-1.32z"
            fill="currentColor"
          />
        </svg>
      </div>
      <h2
        class="label"
      >
        Details
      </h2>
    </header>
    <div
      class="jsx-1555285825 jsx-876534464 "
      data-test="dhis2-uicore-divider"
    />
    <div
      class="children"
      data-test="more-options-children"
    >
      <p>
        Child paragraph
      </p>
    </div>
  </section>
</DocumentFragment>
`;
